lib/puppet_x/puppetlabs/scheduled_task/trigger.rb

```diff
@@ -465,15 +468,27 @@ class Type
       TASK_TRIGGER_SESSION_STATE_CHANGE  = 11
     end
 
-    TYPE_MANIFEST_MAP = {
-      Type::TASK_TRIGGER_DAILY => 'daily',
-      Type::TASK_TRIGGER_WEEKLY => 'weekly',
-      # NOTE: monthly uses context to determine MONTHLY or MONTHLYDOW
-      Type::TASK_TRIGGER_MONTHLY => 'monthly',
-      Type::TASK_TRIGGER_MONTHLYDOW => 'monthly',
-      Type::TASK_TRIGGER_TIME => 'once',
+    TIME_TRIGGER_COLLECTION = {
+      Type::TASK_TRIGGER_DAILY                => 'daily',
+      Type::TASK_TRIGGER_WEEKLY               => 'weekly',
+      # NOTE :monthly uses context to determine MONTHLY or MONTHLYDOW
+      Type::TASK_TRIGGER_MONTHLY              => 'monthly',
+      Type::TASK_TRIGGER_MONTHLYDOW           => 'monthly',
+      Type::TASK_TRIGGER_TIME                 => 'once',
+    }.freeze
+
+    EVENT_TRIGGER_COLLECTION = {
+      # These are event based triggers that will not require a start time.
+      Type::TASK_TRIGGER_BOOT                 => 'boot',
+      Type::TASK_TRIGGER_IDLE                 => 'idle',
+      Type::TASK_TRIGGER_REGISTRATION         => 'registration',
+      Type::TASK_TRIGGER_LOGON                => 'logon',
+      Type::TASK_TRIGGER_SESSION_STATE_CHANGE => 'session_state_change',
+      Type::TASK_TRIGGER_EVENT                => 'event',
     }.freeze
 
+    TYPE_MANIFEST_MAP = (TIME_TRIGGER_COLLECTION.merge(EVENT_TRIGGER_COLLECTION)).freeze
```

Comment:
The current collection defines which trigger types are supported, so I don't think we can add anything here yet without some other code being changed. So I think it might be simpler to just add `boot` to the existing collection for now and handle the one-off differences as we are presently, and tackle the `start_date` / `start_time` problem later.

I'm also not sure if this grouping like this for the sake of finding required values is the right way to go given the chart at https://gist.github.com/Iristyle/58ff446bd3f60e68d52df3868c917a0e#trigger-specific-properties

I had given this a little thought earlier, specifically if we have a way of defining things like optional / required values in the hash from the manifest:

```ruby
triggers = {
Type::TASK_TRIGGER_DAILY                => {
  'name' => 'daily',
  'required' => ['start_time'],
  'optional' => ['start_date', 'end_date', 'minutes_interval', 'minutes_duration', 'every'],
},
Type::TASK_TRIGGER_WEEKLY                => {
  'name' => 'weekly',
  'required' => ['start_time'],
  'optional' => ['start_date', 'end_date', 'minutes_interval', 'minutes_duration', 'every', 'day_of_week'],
},
}
```

This would make the code a lot more self-documenting, as we don't have any sort of clear definition of this mapping in code (aside from mentally unwinding the `canonicalize_and_validate` or similar code).

Reply:
Changed handling boot trigger exemptions to handling the specific case instead of Event Triggers generally.